### PR TITLE
Added: Highlighted staged changes in BulkTagger

### LINKED ISSUE #8658: Bulk Tagger: Highlight staged options
### Describe the problem that you'd like solved

Today, the bulk tagger tool doesn't give any indication of what subjects are staged to be updated.  The only indication that something is staged is the submit button.  If nothing is staged, then the "Submit" button is disabled.  This make it very easy to accidentally add or remove subjects to works.

### Proposal & Constraints

Whenever a subject has been staged for addition or removal, add a highlight to the corresponding menu option.

L9fXjjJRK97eyP6ADySysw2hQguT-rvLXsWYoFN18s9MOFp8m0bL47uUEG104uIsewXPGxef12twoqI3ksyNyKEH_nit396UIoyvCD5DYWWvRZmDaZFPdTQ_cd-mhmPgC2W_-gzHmz0-QpaV7cHqbUw

### Additional context

Bulk tagger menu options are represented by the MenuOption class, in [this file](https://github.com/internetarchive/openlibrary/blob/7e32e7f1bdbb7e2c7ae8408bedd077ee5a2de68f/openlibrary/plugins/openlibrary/js/bulk-tagger/BulkTagger/MenuOption.js).

The [`onMenuOptionClick` function](https://github.com/internetarchive/openlibrary/blob/7e32e7f1bdbb7e2c7ae8408bedd077ee5a2de68f/openlibrary/plugins/openlibrary/js/bulk-tagger/BulkTagger.js#L323) is the called when a menu option is, you guessed it, clicked.

Any additional style rules should be added to [`tagging-menu.less`](https://github.com/internetarchive/openlibrary/blob/7e32e7f1bdbb7e2c7ae8408bedd077ee5a2de68f/static/css/components/tagging-menu.less).

Whenever the bulk tagging form is successfully submitted, the highlighting treatment should be removed from all menu options.

### PR DESCRIPTION
**Pull request by QuantuM410**
Closes #8658 

This PR addresses the issue outlined in #8658 by enhancing the Bulk Tagger tool to provide visual indications of subjects staged for addition or removal. The proposed solution adds highlighting to the corresponding menu options when a subject is staged.

### Technical

The PR integrates the highlighting logic to the staged changes for addition or removal. The highlighting classes are applied based on the updated menu option state. I still have some queries regarding the staged changes as mentioned [**here**](https://github.com/internetarchive/openlibrary/issues/8658#issuecomment-1872234457)

### Testing

1. Open Manage Subjects Menu
image

2. Add / Remove subject to view highlighted staged changes
image

### Screenshot

Before Staging changes
image

After Staging changes
image

### Stakeholders

@jimchamp

### CODE REVIEW

**Comment by QuantuM410** [comment]
@jimchamp Can you please review this PR :)

**Comment by jimchamp** [comment]
If this is ready for review, press the "Ready for review" button.
Is it ready for review?  I said that there should only be yellow highlighting in [this comment](https://github.com/internetarchive/openlibrary/issues/8658#issuecomment-1872552273), but I see both yellow and grey highlighting in your screenshots...

**Comment by QuantuM410** [comment]
> If this is ready for review, press the "Ready for review" button. Is it ready for review? I said that there should only be yellow highlighting in [this comment](https://github.com/internetarchive/openlibrary/issues/8658#issuecomment-1872552273), but I see both yellow and grey highlighting in your screenshots...

Yeh I added grey highlighting to highlight the removed changes and yellow for the addition. I can revert the highlight color for removed to yellow :) @jimchamp

**Comment by QuantuM410** [comment]
@jimchamp I have made this PR ready for review. If you want me to revert the color of "removed" changes to yellow instead, Ill make a commit :+1:

**Comment by QuantuM410** [comment]
@jimchamp Please review it this PR

File: static/css/components/tagging-menu.less [hunk 1]
```diff
@@ -245,6 +245,18 @@
...
+    background-color: @light-grey;
+  }
+
+  &.highlight-some-tagged {
+    background-color: @light-yellow;
+  }
+
+  &.highlight-all-tagged {
+    background-color: @light-yellow;
+  }
```
Comment (jimchamp) [inline static/css/components/tagging-menu.less:259]:
```suggestion
  &--staged {
    background-color: @light-yellow;
  }
```
These menu options will either be staged or not.  Let's use `.selected-tag--staged` to add the yellow background to menu options.

File: openlibrary/plugins/openlibrary/js/bulk-tagger/BulkTagger.js [hunk 2]
```diff
@@ -621,4 +622,32 @@ export class BulkTagger {
...
+            break;
+        case MenuOptionState.ALL_TAGGED:
+            menuOptionElement.classList.remove(
+                'highlight-none-tagged',
+                'highlight-some-tagged'
+            );
+            menuOptionElement.classList.add('highlight-all-tagged');
+            break;
+        }
+    }
```
Comment (jimchamp) [inline openlibrary/plugins/openlibrary/js/bulk-tagger/BulkTagger.js:652]:
```suggestion
```
This much state management is not needed.  Whenever an option is clicked, add the `selected-tag--staged` class.

File: openlibrary/plugins/openlibrary/js/bulk-tagger/BulkTagger.js [hunk 3]
```diff
@@ -345,6 +345,7 @@ export class BulkTagger {
             break
         }
 
+        this.highlightMenuOption(menuOption);
```
Comment (jimchamp) [inline openlibrary/plugins/openlibrary/js/bulk-tagger/BulkTagger.js:348]:
```suggestion
        menuOption.stage()
```
Could you add a `stage` method to `MenuOption`?  This should add the `selected-tag--staged` class to the menu option.

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks @QuantuM410, and apologies for the massive delay.  There's not a need for the level of state management that these changes add --- we need only add a single class if an option is staged.  If a staged option is highlighted, people need only look for the status icon to know if the tag is being added or removed.

I've added some suggested changes that should simplify things a bit.

**Comment by QuantuM410** [comment]
@jimchamp I have implemented the required changes

Before:
Screenshot from 2024-02-22 21-49-30

After:
Screenshot from 2024-02-22 21-49-43

**Pull request merged**